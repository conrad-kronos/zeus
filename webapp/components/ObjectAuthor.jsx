import React, {Component} from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import Gravatar from 'react-gravatar';
import {MdPerson} from 'react-icons/md';

const Avatar = styled.span`
  img {
    border-radius: 2px;
    overflow: hidden;
  }

  img,
  svg {
    margin-right: 5px;
    display: inline-block;
    vertical-align: text-top;
  }
`;

export default class ObjectAuthor extends Component {
  static propTypes = {
    data: PropTypes.shape({
      authors: PropTypes.arrayOf(
        PropTypes.shape({
          name: PropTypes.string,
          email: PropTypes.email
        })
      ).isRequired
    }).isRequired
  };

  render() {
    let {data} = this.props;
    let authors = data.authors;
    if (!authors.length) return null;
    return (
      <span>
        {authors.map(author => {
          if (!author.email || !author.name) return null;
          return (
            <span key={author.email} style={{marginRight: 10}}>
              <Avatar>
                {author.email ? (
                  <Gravatar email={author.email} size={16} />
                ) : (
                  <MdPerson size="16" />
                )}
              </Avatar>
              {author.name || author.email.split('@', 1)[0]}
            </span>
          );
        })}
      </span>
    );
  }
}
