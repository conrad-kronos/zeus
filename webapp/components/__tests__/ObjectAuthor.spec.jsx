import React from 'react';
import {shallow} from 'enzyme';

import ObjectAuthor from '../ObjectAuthor';

describe('ObjectAuthor', () => {
  it('renders author', () => {
    const tree = shallow(
      <ObjectAuthor
        data={{
          authors: [
            {
              email: 'foo@example.com',
              name: 'Foo Bar'
            }
          ]
        }}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders anonymous', () => {
    const tree = shallow(
      <ObjectAuthor
        data={{
          authors: []
        }}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
