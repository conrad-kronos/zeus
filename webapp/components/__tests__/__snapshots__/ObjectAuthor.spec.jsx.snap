// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectAuthor renders anonymous 1`] = `""`;

exports[`ObjectAuthor renders author 1`] = `
<span>
  <span
    key="foo@example.com"
    style={
      Object {
        "marginRight": 10,
      }
    }
  >
    <Avatar>
      <Gravatar
        default="retro"
        email="foo@example.com"
        protocol="//"
        rating="g"
        size={16}
      />
    </Avatar>
    Foo Bar
  </span>
</span>
`;
